src/main/java/com/snowflake/kafka/connector/internal/streaming/TopicPartitionChannel.java

```diff
@@ -1044,6 +1047,9 @@ private SnowflakeStreamingIngestChannel openChannelForTable() {
   public void closeChannel() {
     try {
       this.channel.close().get();
+
+      // telemetry and metrics
+      this.telemetryServiceV2.reportKafkaPartitionUsage(this.snowflakeTelemetryChannelStatus, true);
```

Comment:
Should we send this somewhere periodically instead of just during close?

Reply:
That would be ideal, but currently we don't have anything running periodically in the background. InsertRows is an option, but I'm worried about overloading the telemetry service since we have no control over how often the customer flushes their buffer

Reply:
Is it something we plan to add in the future? I think we send the usage for Snowpipe periodically. Otherwise I feel like this is not very useful since there're no usage reported if there is no open/close on the channel

Reply:
We do not have any plans to add that in the future. It could potentially be a part of the flush service.

Currently we not periodically send Snowpipe usage. `reportKafkaPartition(`) is called on on [close](https://github.com/snowflakedb/snowflake-kafka-connector/blob/bc7af5e54774697d297eb31b4a6980c98cca7588/src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkServiceV1.java#L964) and when the [cleaner runs](https://github.com/snowflakedb/snowflake-kafka-connector/blob/bc7af5e54774697d297eb31b4a6980c98cca7588/src/main/java/com/snowflake/kafka/connector/internal/SnowflakeSinkServiceV1.java#L505), which only runs on the first insert to initialize a pipe